ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -255,6 +255,9 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
             e => {
                 if (e.type === 'DELETED') {
                     setError(new Error('Workflow deleted'));
+                } else if (hasArtifactGCError(e.object.status.conditions)) {
+                    setError(new Error('Artifact garbage collection failed'));
+                    setWorkflow(e.object);
```

Comment:
Nitpick. An alternative way to avoid duplicating the setWorkflow step and thus make the overall logic slightly clearer:
```
else {
    if (hasArtifactGCError(e.object.status.conditions)) {
        setError(new Error('Artifact garbage collection failed'));
    }
    setWorkflow(e.object);
}
```